cocos/2d/assembler/sprite/sliced.ts

```diff
@@ -104,14 +107,31 @@ export const sliced: IAssembler = {
         sizableWidth = sizableWidth < 0 ? 0 : sizableWidth;
         sizableHeight = sizableHeight < 0 ? 0 : sizableHeight;
 
-        dataList[0].x = -appX;
-        dataList[0].y = -appY;
-        dataList[1].x = leftWidth * xScale - appX;
-        dataList[1].y = bottomHeight * yScale - appY;
-        dataList[2].x = dataList[1].x + sizableWidth;
-        dataList[2].y = dataList[1].y + sizableHeight;
-        dataList[3].x = width - appX;
-        dataList[3].y = height - appY;
+        const tempList:IRenderData[] = [];
+        // initialize 4 elements
+        for (let i = 0; i < 4; i++) {
+            tempList.push({ x: 0, y: 0, z: 0, u: 0, v: 0, color: new Color() });
```

Comment:
tempList shouldn't be created inside function logic, try to reuse it for all sliced sprites

Reply:
Modified.